openedx/features/edly/tests/test_utils.py

```diff
@@ -297,3 +303,29 @@ def test_edly_panel_user_has_edly_org_access(self):
         edly_panel_user_group = GroupFactory(name=settings.EDLY_PANEL_USERS_GROUP)
         self.request.user.groups.add(edly_panel_user_group)
         assert edly_panel_user_has_edly_org_access(self.request)
+
+    def test_filter_courses_based_on_org(self):
+        """
+        Test course are filtered properly on current site organization.
+        """
+        edx_org = OrganizationFactory()
+        EdlySubOrganizationFactory(
+            edx_organization=edx_org,
+            lms_site=self.request.site,
+            studio_site=self.request.site
+        )
+        response = cookies_api.set_logged_in_edly_cookies(
+            self.request, HttpResponse(),
+            self.user,
+            cookie_settings(self.request)
+        )
+        self._copy_cookies_to_request(response, self.request)
+        courses = CourseFactory.create_batch(5, org=edx_org.short_name)
+
+        filtered_courses = filter_courses_based_on_org(self.request, courses)
```

Comment:
We need to add test:

1. create course with `org_1`
2. create  a new course with `org_2`
3. Verify that total courses count is 2
4. Link and login the user with `org_1`
5. Verify that `filter_courses_based_on_org` method returns only one course
6. Verify that `returned course org == user's linked edx org`